### DISCUSSION TITLE
Periodic extruded meshes

### DISCUSSION THREAD

**Discussion by golowimmer** [firedrakeproject/firedrake#2570] [Firedrake support]
Hello,
Is there a way to extend the extruded mesh functionality such that the resulting mesh is periodic in the direction of the extrusion? I talked to David last week, and he said there might be a hacky way to do it.

The idea behind this is that I would like to build a tokamak shaped mesh for fusion simulations, with an arbitrary quadrilateral mesh corresponding to a vertical cross-section of the tokamak, which is then extruded and bent around and "glued together" at the ends to arrive at the tokamak mesh.

Unfortunately, given the nature of the problem, it is key to have a mesh whose edges are aligned with the toroidal coordinate (i.e. the long way around the tokamak). At the moment, I am working around this issue by taking a regular quadrilateral cylinder mesh and extruding it radially, but that only allows for rectangular vertical cross-sections.

**Comment by ksagiyam** [comment]
There are many places that need changes in Firedrake and PyOP2, but I think this is doable. Let me give it a try next week.

**Reply by golowimmer** [reply]
Thank you Koki! That would make a big difference to our work.

**Reply by ksagiyam** [reply]
https://github.com/firedrakeproject/firedrake/pull/2582 has been merged!
